eZ/Publish/API/Repository/Values/Content/Query/SortClause/Target/RandomTarget.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\API\Repository\Values\Content\Query\SortClause\Target;
+
+use eZ\Publish\API\Repository\Values\Content\Query\SortClause\Target;
+
+/**
+ * Struct that stores extra target informations for a RandomTarget object.
+ */
+class RandomTarget extends Target
+{
+    /**
+     * @var mixed depends on storage implementation.
+     */
+    public $seed;
+
+    public function __construct($seed)
```

Comment:
@wizhippo this is based on your implementation. Any reason to require API consumer (Developer) to pass `$seed`?

For the Legacy Search Engine, the internal type of `$seed` (int, float) depends on database engine used, so it's kind of design issue for me. Because of that, it should be a responsibility of database engine related strategy to choose seed, not a Developer.

Reply:
@ViniTou let's use here strictly-typed `?int`. To the best of my knowledge, this is most commonly used value of seeds. 

For PostgreSQL implementation (the one that uses float between -1 and 1, right?) please interpolate/normalize `int` value to this range. Sounds a little bit weird, but I feel it's better solution than allowing here mixed.

It should be `null` by default, and if not given computed by either custom implementation or at the moment of using this value (please refrain yourself from computing it in this constructor ;) )